make our home page a bit prettier

andymckay opened the pull request:
- explain a bit more whats going on, i think we should aim to have this as the place we point people too, since they can try it all out
- provide some code examples
- minor nit, remove the product listing code, i don't think that helped and if anyone wants to read the source, its simpler

![screenshot 2015-08-10 16 59 01](https://cloud.githubusercontent.com/assets/74699/9187107/9fca9710-3f81-11e5-8625-92877848861b.png)

Commit:
make our home page a bit prettier

tofumatt commented:
Whoa, that is a LOT more informative! 👍
- tofumatt

On Tue, Aug 11, 2015 at 1:02 AM, Andy McKay notifications@github.com
wrote:

> - explain a bit more whats going on
> - provide some code examples
> - remove the product listing code, i don't think that helped anything
>   ![screenshot 2015-08-10 16 59 01](https://cloud.githubusercontent.com/assets/74699/9187107/9fca9710-3f81-11e5-8625-92877848861b.png)
>   You can view, comment on, or merge this pull request online at:
>   https://github.com/mozilla/payments-example/pull/25
>   -- Commit Summary --
>   - make our home page a bit prettier
>     -- File Changes --
>     A favicon.ico (0)
>     M public/css/index.css (215)
>     A public/fonts/firasans/firasans-bold.eot (0)
>     A public/fonts/firasans/firasans-bold.svg (149)
>     A public/fonts/firasans/firasans-bold.ttf (0)
>     A public/fonts/firasans/firasans-bold.woff (0)
>     A public/fonts/firasans/firasans-bolditalic.eot (0)
>     A public/fonts/firasans/firasans-bolditalic.svg (152)
>     A public/fonts/firasans/firasans-bolditalic.ttf (0)
>     A public/fonts/firasans/firasans-bolditalic.woff (0)
>     A public/fonts/firasans/firasans-book.eot (0)
>     A public/fonts/firasans/firasans-book.svg (149)
>     A public/fonts/firasans/firasans-book.ttf (0)
>     A public/fonts/firasans/firasans-book.woff (0)
>     A public/fonts/firasans/firasans-bookitalic.eot (0)
>     A public/fonts/firasans/firasans-bookitalic.svg (153)
>     A public/fonts/firasans/firasans-bookitalic.ttf (0)
>     A public/fonts/firasans/firasans-bookitalic.woff (0)
>     A public/fonts/firasans/firasans-eight.eot (0)
>     A public/fonts/firasans/firasans-eight.svg (146)
>     A public/fonts/firasans/firasans-eight.ttf (0)
>     A public/fonts/firasans/firasans-eight.woff (0)
>     A public/fonts/firasans/firasans-eightitalic.eot (0)
>     A public/fonts/firasans/firasans-eightitalic.svg (152)
>     A public/fonts/firasans/firasans-eightitalic.ttf (0)
>     A public/fonts/firasans/firasans-eightitalic.woff (0)
>     A public/fonts/firasans/firasans-extrabold.eot (0)
>     A public/fonts/firasans/firasans-extrabold.svg (149)
>     A public/fonts/firasans/firasans-extrabold.ttf (0)
>     A public/fonts/firasans/firasans-extrabold.woff (0)
>     A public/fonts/firasans/firasans-extrabolditalic.eot (0)
>     A public/fonts/firasans/firasans-extrabolditalic.svg (152)
>     A public/fonts/firasans/firasans-extrabolditalic.ttf (0)
>     A public/fonts/firasans/firasans-extrabolditalic.woff (0)
>     A public/fonts/firasans/firasans-extralight.eot (0)
>     A public/fonts/firasans/firasans-extralight.svg (149)
>     A public/fonts/firasans/firasans-extralight.ttf (0)
>     A public/fonts/firasans/firasans-extralight.woff (0)
>     A public/fonts/firasans/firasans-extralightitalic.eot (0)
>     A public/fonts/firasans/firasans-extralightitalic.svg (152)
>     A public/fonts/firasans/firasans-extralightitalic.ttf (0)
>     A public/fonts/firasans/firasans-extralightitalic.woff (0)
>     A public/fonts/firasans/firasans-four.eot (0)
>     A public/fonts/firasans/firasans-four.svg (147)
>     A public/fonts/firasans/firasans-four.ttf (0)
>     A public/fonts/firasans/firasans-four.woff (0)
>     A public/fonts/firasans/firasans-fouritalic.eot (0)
>     A public/fonts/firasans/firasans-fouritalic.svg (151)
>     A public/fonts/firasans/firasans-fouritalic.ttf (0)
>     A public/fonts/firasans/firasans-fouritalic.woff (0)
>     A public/fonts/firasans/firasans-hair.eot (0)
>     A public/fonts/firasans/firasans-hair.svg (147)
>     A public/fonts/firasans/firasans-hair.ttf (0)
>     A public/fonts/firasans/firasans-hair.woff (0)
>     A public/fonts/firasans/firasans-hairitalic.eot (0)
>     A public/fonts/firasans/firasans-hairitalic.svg (153)
>     A public/fonts/firasans/firasans-hairitalic.ttf (0)
>     A public/fonts/firasans/firasans-hairitalic.woff (0)
>     A public/fonts/firasans/firasans-heavy.eot (0)
>     A public/fonts/firasans/firasans-heavy.svg (149)
>     A public/fonts/firasans/firasans-heavy.ttf (0)
>     A public/fonts/firasans/firasans-heavy.woff (0)
>     A public/fonts/firasans/firasans-heavyitalic.eot (0)
>     A public/fonts/firasans/firasans-heavyitalic.svg (152)
>     A public/fonts/firasans/firasans-heavyitalic.ttf (0)
>     A public/fonts/firasans/firasans-heavyitalic.woff (0)
>     A public/fonts/firasans/firasans-italic.eot (0)
>     A public/fonts/firasans/firasans-italic.svg (153)
>     A public/fonts/firasans/firasans-italic.ttf (0)
>     A public/fonts/firasans/firasans-italic.woff (0)
>     A public/fonts/firasans/firasans-light.eot (0)
>     A public/fonts/firasans/firasans-light.svg (149)
>     A public/fonts/firasans/firasans-light.ttf (0)
>     A public/fonts/firasans/firasans-light.woff (0)
>     A public/fonts/firasans/firasans-lightitalic.eot (0)
>     A public/fonts/firasans/firasans-lightitalic.svg (152)
>     A public/fonts/firasans/firasans-lightitalic.ttf (0)
>     A public/fonts/firasans/firasans-lightitalic.woff (0)
>     A public/fonts/firasans/firasans-medium.eot (0)
>     A public/fonts/firasans/firasans-medium.svg (149)
>     A public/fonts/firasans/firasans-medium.ttf (0)
>     A public/fonts/firasans/firasans-medium.woff (0)
>     A public/fonts/firasans/firasans-mediumitalic.eot (0)
>     A public/fonts/firasans/firasans-mediumitalic.svg (153)
>     A public/fonts/firasans/firasans-mediumitalic.ttf (0)
>     A public/fonts/firasans/firasans-mediumitalic.woff (0)
>     A public/fonts/firasans/firasans-regular.eot (0)
>     A public/fonts/firasans/firasans-regular.svg (149)
>     A public/fonts/firasans/firasans-regular.ttf (0)
>     A public/fonts/firasans/firasans-regular.woff (0)
>     A public/fonts/firasans/firasans-regularitalic.eot (0)
>     A public/fonts/firasans/firasans-regularitalic.svg (151)
>     A public/fonts/firasans/firasans-regularitalic.ttf (0)
>     A public/fonts/firasans/firasans-regularitalic.woff (0)
>     A public/fonts/firasans/firasans-semibold.eot (0)
>     A public/fonts/firasans/firasans-semibold.svg (149)
>     A public/fonts/firasans/firasans-semibold.ttf (0)
>     A public/fonts/firasans/firasans-semibold.woff (0)
>     A public/fonts/firasans/firasans-semibolditalic.eot (0)
>     A public/fonts/firasans/firasans-semibolditalic.svg (151)
>     A public/fonts/firasans/firasans-semibolditalic.ttf (0)
>     A public/fonts/firasans/firasans-semibolditalic.woff (0)
>     A public/fonts/firasans/firasans-thin.eot (0)
>     A public/fonts/firasans/firasans-thin.svg (148)
>     A public/fonts/firasans/firasans-thin.ttf (0)
>     A public/fonts/firasans/firasans-thin.woff (0)
>     A public/fonts/firasans/firasans-thinitalic.eot (0)
>     A public/fonts/firasans/firasans-thinitalic.svg (152)
>     A public/fonts/firasans/firasans-thinitalic.ttf (0)
>     A public/fonts/firasans/firasans-thinitalic.woff (0)
>     A public/fonts/firasans/firasans-two.eot (0)
>     A public/fonts/firasans/firasans-two.svg (146)
>     A public/fonts/firasans/firasans-two.ttf (0)
>     A public/fonts/firasans/firasans-two.woff (0)
>     A public/fonts/firasans/firasans-twoitalic.eot (0)
>     A public/fonts/firasans/firasans-twoitalic.svg (149)
>     A public/fonts/firasans/firasans-twoitalic.ttf (0)
>     A public/fonts/firasans/firasans-twoitalic.woff (0)
>     A public/fonts/firasans/firasans-ultra.eot (0)
>     A public/fonts/firasans/firasans-ultra.svg (148)
>     A public/fonts/firasans/firasans-ultra.ttf (0)
>     A public/fonts/firasans/firasans-ultra.woff (0)
>     A public/fonts/firasans/firasans-ultraitalic.eot (0)
>     A public/fonts/firasans/firasans-ultraitalic.svg (153)
>     A public/fonts/firasans/firasans-ultraitalic.ttf (0)
>     A public/fonts/firasans/firasans-ultraitalic.woff (0)
>     A public/fonts/firasans/firasans-ultralight.eot (0)
>     A public/fonts/firasans/firasans-ultralight.svg (148)
>     A public/fonts/firasans/firasans-ultralight.ttf (0)
>     A public/fonts/firasans/firasans-ultralight.woff (0)
>     A public/fonts/firasans/firasans-ultralightitalic.eot (0)
>     A public/fonts/firasans/firasans-ultralightitalic.svg (153)
>     A public/fonts/firasans/firasans-ultralightitalic.ttf (0)
>     A public/fonts/firasans/firasans-ultralightitalic.woff (0)
>     M public/index.html (160)
>     M public/js/index.js (74)
>     -- Patch Links --
>     https://github.com/mozilla/payments-example/pull/25.patch
>     https://github.com/mozilla/payments-example/pull/25.diff
>     ---
>     Reply to this email directly or view it on GitHub:
>     https://github.com/mozilla/payments-example/pull/25

muffinresearch commented on public/index.html:
```diff
@@ -1,54 +1,124 @@
+    </section>
+
+    <section>
+      <h2><a name="drop-in"></a>Drop-in UI</h2>
```
Maybe we could call this something different. E.g. payments client? I thought this mean the braintree drop-in UI initially.

muffinresearch commented on public/index.html:
```diff
@@ -1,54 +1,124 @@
+    </section>
+
+    <section>
+      <h2><a name="full-api"></a>Full API</h2>
```
If you just want headings to be referenced with a fragment identifier e.g: #full-api `<h1 id="full-api">Full API</h1>` will give you that you don't need the link.

muffinresearch commented:
Maybe constrain the text so it can't get too wide. It looks slightly odd on a wide screen:

![payments_example](https://cloud.githubusercontent.com/assets/1514/9193830/72b9ef9a-400e-11e5-8264-b46a74a93742.png)

To do that it looks like you can just stick a `<div class="container">` around the content in `<div class="callout" />`

muffinresearch commented:
Below a certain width  (~838px)  the nav text dissappears: 

![payments_example](https://cloud.githubusercontent.com/assets/1514/9193970/b33310be-400f-11e5-9828-d732b86c65e4.png)

muffinresearch commented:
Looks a million miles better than the old example!

r+wc

muffinresearch commented:
Now I'm running this branch I notice this is missing the management button.

Commit:
feedback from @muffinresearch and @tofumatt

andymckay commented:
- all the feedback fixed, thanks
- improved the nav a little
- quick example seemed pointless, moved the button up into the callout
- added contact us section

<img width="961" alt="screenshot 2015-08-11 10 53 22" src="https://cloud.githubusercontent.com/assets/74699/9205514/84f60dd4-4017-11e5-9c8c-a0e8bb5469d8.png">

kumar303 commented on public/index.html:
```diff
@@ -1,54 +1,139 @@
+  <div class="callout">
+      <div class="container">
+    <p><a href="https://payments.readthedocs.org">Mozilla Payments</a> is a unified
+      payments interface to Mozilla sites and properties. We provide a
```
Very concise

kumar303 commented on public/index.html:
```diff
@@ -1,54 +1,139 @@
+      payments interface to Mozilla sites and properties. We provide a
+      wrapper around <a href="https://www.braintreepayments.com/">Braintree</a>.
+      Consider us a replacement for using Stripe, Braintree and others.</p>
+    <p>Work at the Mozilla Foundation or Mozilla Corporation and want to accept payments? This is for you.</p>
```
Is there a way to rephrase this so we can focus on the products rather than the developers? Maybe something like: "Are you looking to accept payments on a Mozilla Foundation or Mozilla Corporation site?"

andymckay replied:
i like that phrasing

kumar303 commented on public/index.html:
```diff
@@ -1,54 +1,139 @@
+        <li>Subscriptions (including handling of failures and retries)</li>
+        <li>Management interface</li>
+        <li>Emails and receipts</li>
+        <li>Localisations</li>
```
You can add 'secure credit card processing' to the list

kumar303 commented on public/index.html:
```diff
@@ -1,54 +1,139 @@
+      <p><a href="http://payments.readthedocs.org/en/latest/design/api/configuration.html">read more →</a></p>
+
+      <h3>2. Button code</h3>
+      <p>Add in the <a href="http://payments.readthedocs.org/en/latest/design/api/client.html">button code</a> to trigger. In the above example, it listens to click events on a button and trigger the following:</p>
```
might read better as: "...to trigger the payment screen"

kumar303 commented on public/index.html:
```diff
@@ -1,54 +1,139 @@
+        <p>Find us: #payments on IRC, at <a href="https://mail.mozilla.org/listinfo/dev-payments">dev-payments</a> or on <a href="https://github.com/mozilla/payments">github</a>. You could also contact Andy McKay or Lindsay Saunders.</p>
+    </section>
+
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
do we need both jquerys? We are already loading `lib/js/jquery.min.js`

andymckay replied:
doh

kumar303 commented on public/js/index.js:
```diff
@@ -47,129 +48,88 @@
+  });
+
+  $('button.management').on('click', function(event) {
+    window.location = config.manageUrl;
```
This should log in first, then redirect to the management app with a token. The old code did this:

```
fxaSignIn()
  .then(function(result) {
    window.location.href = config.manageUrl + '?access_token=' + result.access_token;
  });
```

We were mainly just using this as a way to test token sign-ins to management since you can already test user/password sign-ins by directly hitting the management URL (i.e. without a token).

kumar303 commented:
Nice! This should be helpful for people. I added a few comments.

andymckay commented:
Good calls, thanks kumar.

Commit:
feedback from kumar
